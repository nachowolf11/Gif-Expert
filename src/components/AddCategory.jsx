import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const inputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSumbit = (e) => {
        e.preventDefault()
        if( inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim()) // Esto serÃ­a como esto: onAddCategory(inputValue.trim())
        setInputValue('')
    }

  return (
    <form onSubmit={onSumbit} aria-label="form">
        <input
            type="text"
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={(event)=> inputChange(event)}
        />
    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}