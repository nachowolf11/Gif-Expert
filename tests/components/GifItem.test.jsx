import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas de GiftItem', () => {
    
    const title = 'Titulo1';
    const url = 'https//url.com';
    
    test('Debe coincidir el snapshot ', () => {
        const { container } = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    });
});