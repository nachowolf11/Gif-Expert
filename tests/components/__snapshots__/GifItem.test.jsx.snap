// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GiftItem Debe coincidir el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo1"
      src="https//url.com"
    />
    <p>
      Titulo1
    </p>
  </div>
</div>
`;
